src/sidebar/components/share-annotations-panel.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+const { useRef } = require('preact/hooks');
+const useStore = require('../store/use-store');
+const { copyText } = require('../util/copy-to-clipboard');
+const { withServices } = require('../util/service-context');
+const uiConstants = require('../ui-constants');
+
+const SidebarPanel = require('./sidebar-panel');
+const SvgIcon = require('./svg-icon');
+
+/**
+ * A panel for sharing the current group's annotations.
+ *
+ * Links withinin this component allow a user to share the set of annotations that
+ * are on the current page (as defined by the main frame's URI) and contained
+ * within the app's currently-focused group.
+ */
+function ShareAnnotationsPanel({ analytics, flash }) {
+  const mainFrame = useStore(store => store.mainFrame());
+  const focusedGroup = useStore(store => store.focusedGroup());
+
+  // We can render a basic frame for the panel at any time,
+  // but hold off rendering panel content if needed things aren't present.
+  // We need to know what page we're on and what group is focused.
+  const shouldRenderPanelContent = focusedGroup && mainFrame;
+
+  const groupName = (focusedGroup && focusedGroup.name) || '...';
+  const panelTitle = `Share Annotations in ${groupName}`;
+
+  // Generate bouncer sharing link for annotations in the current group
+  const shareURI = ((frame, group) => {
+    if (!frame || !group) {
+      return '#';
+    }
+    return `https://hyp.is/go?url=${encodeURIComponent(mainFrame.uri)}&group=${
+      group.id
+    }`;
+  })(mainFrame, focusedGroup);
+
+  const trackShareClick = shareTarget => {
+    analytics.track(analytics.events.DOCUMENT_SHARED, shareTarget);
+  };
+
+  const copyShareLink = () => {
+    try {
+      copyText(shareURI);
+      flash.info('Copied share link to clipboard');
+    } catch (err) {
+      flash.error('Unable to copy link');
+    }
+  };
+
+  const inputRef = useRef();
+
+  const focusShareLink = () => {
```

Comment:
I noticed this is getting fired twice when we click on the input, probably because we have onClick and onFocus. Note: My the debugger did not catch twice, but logging will.

I'm wondering if we can't just use onFocus only? Or `event.stopPropagation()` might work from the event param.

Reply:
I might yoink this entirely until the next round of PRs in which we can really drill down on the focus behavior overall as it's something I wanted to improve in general.